services/common/src/redux/customAxios.js

```diff
@@ -15,6 +19,61 @@ const formatErrorMessage = (errorMessage) => {
   return errorMessage.replace("(psycopg2.", "(DatabaseError.");
 };
 
+const decodeJWT = (token) => {
+  var base64Url = token.split('.')[1];
+  var base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
+  var jsonPayload = decodeURIComponent(window.atob(base64).split('').map(function(c) {
+      return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
+  }).join(''));
+
+  return JSON.parse(jsonPayload);
+};
+
+const notifymAdmin = (error) => {
+
+  const business_message = error?.response?.data?.message;
+  const detailed_error = error?.response?.data?.detailed_error;
+  let date = new Date()
+  const reported_date = `${date} ${date.getHours()}:${date.getMinutes()}`;
+  const user = decodeJWT(createRequestHeader().headers.Authorization)
+  const environment_name = ENVIRONMENT.environment.toString().toUpperCase();
+
+  const email_title = "[MDS_INCIDENT_REPORT] [" + environment_name + "] - " + reported_date + " - " + business_message;
+  const email_body = `<p><b>Environment:</b> ${environment_name}</P>
+    <p><b>Business Error:</b> ${business_message}</P>
+    <p>
+      <b>Reporter's Name:</b> ${user.given_name} ${user.family_name}</br>
+      <b>Reporter's Email:</b> ${user.email}</br>
```

Comment:
With the change suggested above to make the email endpoint specific to error reporting, I would suggest trying to get these details (user email / username etc) from the users token on the endpoint level so you would not be able to send these on behalf of another user

Reply:
This is addressed. Thanks.